btrdbextras/eventproc/eventproc.py

```diff
@@ -211,3 +214,48 @@ def inner(func):
         return func
 
     return inner
+
+def upload_file(file, file_name):
+    """
+    Uploads file to S3. Returns a link to download the file.
+    If the function runs outside of an eventproc handler executing in response to a hook, it will just check the inputs, raise a warning, and return None.
+
+    Parameters
+    ----------
+    file: string
+        Path to the file.
+    file_name: string
+        Name that the file will be called on download. Maximum 36 characters.
+
+    Raises
+    ---------
+    TypeError: file must be a string.
+    TypeError: file_name must be a string.
+    ValueError: file must be a path to a file.
+    ValueError: file_name cannot be longer than 32 characters, is <actual length>.
+    
+    Returns
+    ----------
+    string: Download link to the object. None if upload was not attempted.
+    """
+    
+    # check the inputs
+    if not isinstance(file, str):
+        raise TypeError("file must be a string.")
+    if not isinstance(file_name, str):
+        raise TypeError("file_name must be a string.")
+    if not os.path.exists(file):
+            raise ValueError("file must be a path to a file.")
+    if len(file_name) > 32:
+        raise ValueError("file_name cannot be longer than 32 characters, is {0}.".format(len(file_name)))
+
+    # check the context
+    if not os.getenv("EXECUTOR_CONTEXT") == "true":
+        warnings.warn("upload_file is running in an execution context without the appropriate AWS credentials and will not upload to S3.")
+        return None
+        
+    # queue the upload, to be completed by the executor when the handler completes
+    code = str(uuid.uuid4().hex)
+    _uploads[code] = [os.getenv("HOME") + "/" + file, file_name]
+        
+    return "https://downloads.{0}/{1}".format(os.getenv("CLUSTER_NAME"), code)
```

Comment:
I think this should be a computed value under endpoints in the siteconfig.  

It should also get added to certmanager although this can be done in the follow-on ticket to make the downloader.  

You can leave this code as is if you make sure to update when creating the download service.